src/main/java/com/scalar/db/storage/jdbc/query/InsertQuery.java

```diff
@@ -0,0 +1,93 @@
+package com.scalar.db.storage.jdbc.query;
+
+import com.scalar.db.io.Key;
+import com.scalar.db.io.Value;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class InsertQuery extends AbstractQuery {
+
+  public static class Builder {
+    private final String fullTableName;
+    private Key partitionKey;
+    private Optional<Key> clusteringKey;
+    private Map<String, Value> values;
+
+    Builder(String fullTableName) {
+      this.fullTableName = fullTableName;
+    }
+
+    public Builder values(
+        Key partitionKey, Optional<Key> clusteringKey, Map<String, Value> values) {
+      this.partitionKey = partitionKey;
+      this.clusteringKey = clusteringKey;
+      this.values = values;
+      return this;
+    }
+
+    public InsertQuery build() {
+      return new InsertQuery(this);
+    }
+  }
+
+  private final String fullTableName;
```

Comment:
It might be just a style thing (not sure if google's style guide defines something for this style), but is it common to instance variables in the middle if some static inner class comes above?

Reply:
Not sure it's common or not actually, but my policy was that static fields and methods were placed first in a class. Let me check google's style guide.

Reply:
As we discussed, it would be great if we can follow [this](https://github.com/scalar-labs/scalar/blob/master/common/src/main/java/com/scalar/dl/ledger/asset/AssetProof.java) for consistency.